### ISSUE TITLE
shuffle=True??

### ISSUE THREAD

**Issue by asehmi** [chibui191/bitcoin_volatility_forecasting#2]
Dear Chi Bui,

I came across this repo via ml-quant.com and was very pleased to see the full unfolding of your BTC price prediction experiments, from GARCH to multi-variate LSTM! Great job, and thank you.

I have done something similar with LSTMs and am working with both BTC-USD and ETH-USD data to build a short-term price forecast model for ETH-USD. The model uses various financial technical indicators (such as EMA, OBV, RSI) and sentiment. In my model I was careful to use `shuffle=False` in `model.fit()`, whereas I see you are using `shuffle=True`. I was under the impression this is not recommended for time series. Can you explain why you did this? Wouldn't it be problematic for the `Bidirectional` layer?

Did you experiment with different `loss`, `activation` and `optimizer` functions before arriving at your selection?

Many thanks,
Arvindra

**Comment by chibui191** [comment]
Hello @asehmi, thank you for reaching out.

Actually setting `shuffle=True` is a practice that I picked up from the [Sequences, Time Series and Prediction course by DeepLearning.AI](https://www.coursera.org/learn/tensorflow-sequences-time-series-and-prediction). The short explanation to this is that the shuffle happens on the batches axis, not the time. The `windowed_dataset` function that I created above helped split the data into smaller sequences, and the order in which these sequences being fed into the NN is what's shuffled. You can also find more details on this [here](https://stackoverflow.com/questions/57041305/keras-shuffling-dataset-while-using-lstm).

Thank you for studying my project so closely and catching such small detail.
Chi

**Comment by asehmi** [comment]
Thank you. Understood. Very grateful for the pointers!

**Comment by chibui191** [comment]
Great! I'm glad you find it helpful. Good luck on your project!
